### PR TITLE
Channels2

### PR DESCRIPTION
**Pull request by e-stoj**

### CODE REVIEW

File: src/main/java/com/example/chat/groupchatbackend/controllers/ConversationController.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
```
Comment (michal-marciniec) [inline src/main/java/com/example/chat/groupchatbackend/controllers/ConversationController.java]: please add also `@CrossOrigin(origins = "*", allowCredentials = "true", maxAge = 3600L)`

File: src/main/java/com/example/chat/groupchatbackend/controllers/ConversationController.java [hunk 2]
```diff
@@ -39,5 +36,41 @@ public ResponseEntity createChannel(@RequestBody String name) {
...
+
+    @GetMapping("/direct-messages")
+    public List<BasicConversation> getAllDirectMessages() {
+        return getBasicConversations(ConversationType.DIRECT_MESSAGE);
+    }
+
+    private List<BasicConversation> getBasicConversations(ConversationType conversationType) {
+        Iterable<Conversation> channelsIterable = conversationsRepository.findAll();
+        List<Conversation> channels = (List<Conversation>) channelsIterable;
+        return channels.stream()
```
Comment (michal-marciniec) [inline src/main/java/com/example/chat/groupchatbackend/controllers/ConversationController.java]: you can add method `findByConversationType(ConversationType type)` to `conversationRepository` and use it here :) you even don't need to implement it, Spring will generate sql itself basing on method name

File: src/main/java/com/example/chat/groupchatbackend/controllers/ConversationController.java [hunk 3]
```diff
@@ -1,19 +1,16 @@
...
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
```
Comment (michal-marciniec) [inline src/main/java/com/example/chat/groupchatbackend/controllers/ConversationController.java:13]: plase add the following line to the class `@CrossOrigin(origins = "*", allowCredentials = "true", maxAge = 3600L)`

**Review by michal-marciniec — COMMENTED** [review]
1 change and 1 line to add, and will be good to go :)

**Review by michal-marciniec — APPROVED** [review]

**Review by mateusz-siwiec — APPROVED** [review]

**Pull request merged**